spacelift/resource_stack_dependency.go

```diff
@@ -0,0 +1,152 @@
+package spacelift
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func resourceStackDependency() *schema.Resource {
+	return &schema.Resource{
+		Description: "`spacelift_stack_dependency` represents a Spacelift **stack dependency** - " +
+			"a dependency between two stacks. When one stack depends on another, the tracked runs " +
+			"of the stack will not start until the dependent stack is successfully finished.",
+
+		CreateContext: resourceStackDependencyCreate,
+		ReadContext:   resourceStackDependencyRead,
+		UpdateContext: resourceStackDependencyUpdate,
+		DeleteContext: resourceStackDependencyDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceStackDependencyImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"stack_id": {
+				Type:        schema.TypeString,
+				Description: "immutable ID (slug) of stack which has a dependency.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"depends_on_stack_id": {
+				Type:        schema.TypeString,
+				Description: "immutable ID (slug) of stack to depend on.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"triggers": {
+				Type:        schema.TypeBool,
+				Description: "describes whether we should trigger the dependent if it's not triggered by the push, but the current stack has changed. Defaults to `true`.",
+				Optional:    true,
+				Default:     true,
+			},
+		},
+	}
+
+}
+
+func resourceStackDependencyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		StackDependency structs.StackDependency `graphql:"stackDependencyCreate(input: $input)"`
+	}
+
+	variables := map[string]interface{}{"input": stackDependencyCreateInput(d)}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "StackDependencyCreate", &query, variables); err != nil {
+		return diag.Errorf("could not create stack dependency: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s|%s", query.StackDependency.ID, query.StackDependency.StackID))
```

Comment:
Why does this use the `|` character, instead of using `/` like the backend (and i.e. the policy attachment resource)?

Reply:
because there is a slash in the dependency slug 😂

Reply:
whaaaat, how so?

Reply:
Slashes can be in slugs? Wouldn't that break the frontend URLs?

Reply:
<img width="388" alt="image" src="https://user-images.githubusercontent.com/19969687/187216891-4b9f2546-0a6d-4203-9beb-962ddb46f0f5.png">